File: Oct2017/SolutionFindDisappearedNumbers.java
class SolutionFindDisappearedNumbers {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> ans = new ArrayList<Integer>();
        for(int i = 0; i < nums.length; i++){
            int idx = Math.abs(nums[i]) - 1;
            if(nums[idx] > 0) nums[idx] = -nums[idx];
        }
        for(int i = 0; i < nums.length; i++){
            if(nums[i] > 0) ans.add(i+1);
        }
        
        return ans;
    }
}

File: Oct2017/SolutionFindDisappearedNumbers.py
class SolutionFindDisappearedNumbers(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = []
        for val in nums:
            idx = abs(val) - 1
            if nums[idx] > 0:
                nums[idx] = -nums[idx]
            
        for idx, val in enumerate(nums):
            if val > 0:
                ans.append(idx + 1)
        
        return ans
            
            
        